planning/behavior_path_start_planner_module/src/start_planner_module.cpp

```diff
@@ -676,6 +660,31 @@ bool StartPlannerModule::findPullOutPath(
   return true;
 }
 
+PathWithLaneId StartPlannerModule::extractCollisionCheckPath(const PullOutPath & path)
+{
+  PathWithLaneId combined_path;
+  for (const auto & partial_path : path.partial_paths) {
+    combined_path.points.insert(
+      combined_path.points.end(), partial_path.points.begin(), partial_path.points.end());
+  }
+  // calculate collision check end idx
+  const size_t collision_check_end_idx = std::invoke([&]() {
+    const auto collision_check_end_pose = motion_utils::calcLongitudinalOffsetPose(
+      combined_path.points, path.end_pose.position, parameters_->collision_check_distance_from_end);
+
+    if (collision_check_end_pose) {
+      return motion_utils::findNearestIndex(
+        combined_path.points, collision_check_end_pose->position);
+    } else {
+      return combined_path.points.size() - 1;
```

Comment:
```suggestion
      return 0;
```
is reasonable?

Reply:
or nullopt?

Reply:
changed in https://github.com/autowarefoundation/autoware.universe/pull/5982/commits/6839191456f0dc84228f824dd4dbc692d48d20b3